Sources/StreamChat/ChatClient.swift

```diff
@@ -299,8 +299,10 @@ public class _ChatClient<ExtraData: ExtraDataTypes> {
         self.eventWorkerBuilders = eventWorkerBuilders
 
         currentUserId = fetchCurrentUserIdFromDatabase()
-
-        clientUpdater.reloadUserIfNeeded(completion: completion)
+        // Backward compatiability
+        if let tokenProvider = tokenProvider {
+            setConnectionInfoAndConnect(tokenProvider: tokenProvider, completion: completion)
```

Comment:
@VojtaStavik  This part is controversial, on the one hand we need it for backward compatibility, on the other hand it's now possible to both pass a tokenProvider provider, and then connect once again by calling one of the `connect` methods. It doesn't break anything, plus you can always just call `connect` multiple times, so maybe it's ok after all